src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -357,22 +415,27 @@
 
 (def fact-paths-query
   "Query for the resource-params query, mostly used as a subquery"
-  (map->Query {:projections {"type" {:type :string
+  (map->Query {:can-drop-unused-joins? true
+               :projections {"type" {:type :string
                                      :queryable? true
-                                     :field :vt.type}
+                                     :field :vt.type
+                                     :join-deps #{:vt}}
                              "path" {:type :path
                                      :queryable? true
-                                     :field :path}
+                                     :field :path
+                                     :join-deps #{:fp}}
                              "name" {:type :string
                                      :queryable? true
-                                     :field :name}
+                                     :field :name
+                                     :join-deps #{:fp}}
                              "depth" {:type :numeric
                                       :queryable? true
                                       :query-only? true
-                                      :field :fp.depth}}
+                                      :field :fp.depth
+                                      :join-deps #{:fp}}}
                :selection {:from [[:fact_paths :fp]]
-                           :join [[:value_types :vt]
-                                  [:= :fp.value_type_id :vt.id]]
+                           :left-join [[:value_types :vt]
```

Comment:
Is this switching the existing query from an inner join to a left-join? Does that matter in this case? I think we talked about it before but can't remember.

Reply:
Yes, and I think no?  i.e. I *think* it's fine for it to be a left join since we're just adding a type for each "from" path, and there better always be a fact type?